### PR TITLE
YAML metrics configuration handling for C++ backend

### PR DESCRIPTION
**Pull request by namannandan**
## Description

This PR includes the implementation for YAML metrics configuration handling for C++ backend.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Feature/Issue validation/testing

Unit tests for various configuration file content scenarios are included in the PR.

## Reference design
- https://github.com/pytorch/serve/issues/1492
- [Implementation details](https://docs.google.com/document/d/1DrInbQx7SFP9YG2xPS7Ca6s6qV5hf9_cjm7wic5RZP0/edit#heading=h.zhozrmkjz4e7)

### CODE REVIEW

**Comment by namannandan** [comment]
Prior PR https://github.com/namannandan/serve/pull/1 with relevant code changes and review comments which have been addressed here.

File: cpp/build.sh [hunk 1]
```diff
@@ -358,5 +391,6 @@ cd "$(dirname "$0")"
 install_folly
 install_kineto
 install_libtorch
+install_yaml_cpp
```
[the remark this answers is not in the document]
Comment (namannandan) [reply cpp/build.sh:402]: Done

File: cpp/build.sh [hunk 2]
```diff
@@ -201,6 +201,39 @@ function install_libtorch() {
   cd "$BWD" || exit
 }
 
+function install_yaml_cpp() {
+  YAML_CPP_SRC_DIR=$BASE_DIR/third-party/yaml-cpp
+  YAML_CPP_BUILD_DIR=$YAML_CPP_SRC_DIR/build
```
Comment (lxning) [inline cpp/build.sh:206]: The build dir is not correct. It should be under dir _build/, not third-party dir. Please ref https://github.com/pytorch/serve/blob/cpp_backend/cpp/build.sh#L108

File: cpp/build.sh [hunk 3]
```diff
@@ -290,6 +323,14 @@ function symlink_torch_libs() {
   fi
 }
 
+function symlink_yaml_cpp_lib() {
+  if [ "$PLATFORM" = "Linux" ]; then
+    ln -sf ${BASE_DIR}/third-party/yaml-cpp/build/*.so* ${BUILD_DIR}/libs/
```
Comment (lxning) [inline cpp/build.sh:328]: the ${BASE_DIR}/third-party/yaml-cpp/build/*.so* is not correct.

File: cpp/build.sh [hunk 4]
```diff
@@ -290,6 +323,14 @@ function symlink_torch_libs() {
...
+  elif [ "$PLATFORM" = "Mac" ]; then
+    ln -sf ${BASE_DIR}/third-party/yaml-cpp/build/*.dylib* ${BUILD_DIR}/libs/
```
Comment (lxning) [inline cpp/build.sh:330]: same here.

**Comment by namannandan** [comment]
Addressed review comments. CI is passing. Merging PR.

**Pull request merged**